raiden/tests/unit/test_channelstate.py

```diff
@@ -1910,6 +1927,257 @@ def test_receive_withdraw_confirmation():
     )
 
     assert (
-        search_for_item(iteration.events, ContractSendChannelWithdraw, {"total_withdraw": 70})
+        search_for_item(
+            iteration.events, ContractSendChannelWithdraw, {"total_withdraw": total_withdraw}
+        )
+        is not None
+    )
+
+
+def test_node_sends_withdraw_expiry():
+    pseudo_random_generator = random.Random()
+
+    our_model1, _ = create_model(balance=70)
+    partner_model1, privkey2 = create_model(balance=100)
+    channel_state = create_channel_from_models(our_model1, partner_model1, privkey2)
+    block_hash = make_block_hash()
+
+    total_withdraw = 50
+    expiration_block_number = 10
+    expiration_threshold = channel.get_sender_expiration_threshold(expiration_block_number)
+
+    channel_state.our_state.total_withdraw = total_withdraw
+    channel_state.our_state.withdraws.append(
+        WithdrawState(total_withdraw=total_withdraw, expiration=expiration_block_number, nonce=1)
+    )
+
+    block_hash = make_transaction_hash()
+    block = Block(block_number=expiration_threshold - 1, gas_limit=1, block_hash=block_hash)
+
+    iteration = channel.handle_block(
+        channel_state=channel_state,
+        state_change=block,
+        block_number=expiration_threshold - 1,
+        pseudo_random_generator=pseudo_random_generator,
+    )
+
+    assert iteration.events == []
+
+    block_hash = make_transaction_hash()
+    block = Block(block_number=expiration_threshold, gas_limit=1, block_hash=block_hash)
+
+    iteration = channel.handle_block(
+        channel_state=channel_state,
+        state_change=block,
+        block_number=expiration_threshold,
+        pseudo_random_generator=pseudo_random_generator,
+    )
+
+    assert (
+        search_for_item(
+            iteration.events,
+            SendWithdrawExpired,
+            {
+                "total_withdraw": total_withdraw,
+                "participant": channel_state.our_state.address,
+                "recipient": channel_state.partner_state.address,
+                "nonce": 1,
```

Comment:
humm, this cannot be right, how can the nonce be 1? at least a withdraw had to be sent before.

Edit: This is serious, it looks like we have multiple copies of the nonce and they are not in sync, IMO this is a design flaw and should be fixed.

Reply:
[The withdraw state](https://github.com/raiden-network/raiden/pull/4289/files#diff-f7d7810957ef3a6c3153128196457a74R1951) created was provided with a nonce of (1). That is, similar in structure to the `BalanceProof` states which store the nonce.
This would be the value compared against the next expected nonce which in this case, would be 1.